lalrpop/src/normalize/token_check/mod.rs

```diff
@@ -21,19 +21,50 @@ mod test;
 
 pub fn validate(mut grammar: Grammar) -> NormResult<Grammar> {
     let (has_enum_token, all_literals) = {
+        let opt_match_token = grammar.match_token();
+
+        let (match_mappings, match_catch_all) = if let Some(mt) = opt_match_token {
```

Comment:
I wonder if we can find a name for `match_mappings` that makes it clear what *direction* the map takes place. After all, the user writes `r"foo" => "bar"`, but this table is going in the inverse direction. Maybe we can call it `match_rev_mapping`? Or `user_name_to_match_map`? Something like that...